daemon/daemon.go

```diff
@@ -377,6 +377,11 @@ func (d *Daemon) updatePolicy(spec update.Spec, updates policy.Updates) updateFu
 		var anythingAutomated bool
 
 		for serviceID, u := range updates {
+			if d.Cluster.IsAllowedResource(serviceID) {
```

Comment:
This may be where my unease at looking at resource IDs comes from -- this is a user-provided value, and we're trusting what it says. Maybe harmless now, but a potential loophole.

Reply:
Well, it's comparing against yet another user-provided value (since it comes from the user).

However, for extra assurance, we could be calling or using the output of `getAllowedNamespaces` in `IsAllowedResource()` so that we check against existing namespaces. I don't like this either though, since you may want to restrict the view to namespaces you haven't created yet.

Reply:
In fact, for performance and simplicity, I was pondering to remove the existing calls to `getAllowedNamespaces()` by direct access to the provided allowed namespaces when possible (i.e. everywhere except `Export()`).

Reply:
I see the reasoning here, sometimes you want `{ namespaces named as allowed }`; e.g., when creating a namespace from a manifest. And sometimes you want `{ namespaces allowed and existing }`; e..g, when querying for cluster workloads.